RIOT-OS/RIOT
# gnrc_rpl_p2p: several compilation fixes

### Contribution description
A couple of issues appear when compiling `gnrc_rpl_p2p` (reported in #13369).
This PR fixes the cleanup timer usage and uses the correct netif functions.

### Testing procedure
* add `USEMODULE += gnrc_rpl_p2p` to `examples/gnrc_networking`.
* start two or more native instances
* add a global address to one native instance `ifconfig 6 add abcd::1` (neighbour discovery propagates a prefix to the other native instances).
* start p2p-rpl on the node that configured `abcd::1` to find a path to one of the native instances (assuming a global address of `abcd::bc63:a1ff:fe23:df98`):
`rpl find abcd::1 abcd::bc63:a1ff:fe23:df98`
* `rpl show` lists a temporary dodag
* `nib route` should show a forwarding entry to the second native instance

### Issues/PRs references
fixes #13369
